File: app/src/main/java/khorunzhyicom/alex/theposter/service/models/Movie.java
package khorunzhyicom.alex.theposter.service.models;


import android.support.annotation.Nullable;

import org.immutables.value.Value;

import java.util.Date;

@Value.Immutable
public interface Movie {

    String title();
    String originalTitle();
    @Nullable Date releaseDate();
    @Nullable Integer budget();
    @Nullable Integer voteCount();
    @Nullable Double voteAverage();
    @Nullable Double popularity();
    @Nullable String backdropPath();
    @Nullable String overview();
    @Nullable String posterPath();
}

File: app/src/main/java/khorunzhyicom/alex/theposter/service/models/Movie.kt
package khorunzhyicom.alex.theposter.service.models


import java.util.*

data class Movie (
        val title: String,
        val originalTitle: String,
        val releaseDate: Date?,
        val budget: Int?,
        val voteCount: Int?,
        val voteAverage: Double?,
        val popularity: Double?,
        val backdropPath: String?,
        val overview: String?,
        val posterPath: String?
)
